cell2location/distributions/AutoAmortisedNormalMessenger.py

```diff
@@ -0,0 +1,349 @@
+            x_in = [kwargs[i] if i in kwargs.keys() else args[i] for i in in_names]
+            # apply data transform before passing to NN
+            in_transforms = self.amortised_plate_sites["input_transform"]
+            x_in = [in_transforms[i](x) for i, x in enumerate(x_in)]
+            if "single" in self.encoder_mode:
+                # encode with a single encoder
+                res = deep_getattr(self, "one_encoder")(*x_in)
+                if "multiple" in self.encoder_mode:
+                    # when there is a second layer of multiple encoders fetch encoders and encode data
+                    x_in[0] = res
+                    res = deep_getattr(self.multiple_encoders, name)(*x_in)
+            else:
+                # when there are multiple encoders fetch encoders and encode data
+                res = deep_getattr(self.multiple_encoders, name)(*x_in)
+            return res
+        except AttributeError:
+            pass
+
+        # Initialize.
+        # create single encoder NN
+        if "single" in self.encoder_mode:
+            if self.encoder_instance is not None:
+                # copy provided encoder instance
+                one_encoder = deepcopy(self.encoder_instance).to(prior.mean.device)
+                # convert to pyro module
+                to_pyro_module_(one_encoder)
+                deep_setattr(self, "one_encoder", one_encoder)
+            else:
+                # create encoder instance from encoder class
+                deep_setattr(
+                    self,
+                    "one_encoder",
+                    self.encoder_class(n_in=self.single_n_in, n_out=self.n_hidden["single"], **self.encoder_kwargs).to(
+                        prior.mean.device
+                    ),
+                )
+        if "multiple" in self.encoder_mode:
+            # determine the number of hidden layers
+            if name in self.n_hidden.keys():
+                n_hidden = self.n_hidden[name]
+            else:
+                n_hidden = self.n_hidden["multiple"]
+            multi_encoder_kwargs = deepcopy(self.multi_encoder_kwargs)
+            multi_encoder_kwargs["n_hidden"] = n_hidden
+
+            # create multiple encoders
+            if self.encoder_instance is not None:
+                # copy instances
+                encoder_ = deepcopy(self.encoder_instance).to(prior.mean.device)
+                # convert to pyro module
+                to_pyro_module_(encoder_)
+                deep_setattr(
+                    self,
+                    "multiple_encoders." + name,
+                    encoder_,
+                )
+            else:
+                # create instances
+                deep_setattr(
+                    self,
+                    "multiple_encoders." + name,
+                    self.encoder_class(n_in=self.multiple_n_in, n_out=n_hidden, **multi_encoder_kwargs).to(
+                        prior.mean.device
+                    ),
+                )
```

Comment:
missing a return statement in the initialization branch?

Reply:
Fixed this now. I wonder why it worked without that return statement. Probably it worked because when the amortized NN does not exist, pyro parameters for converting the hidden layers to loc/scales/weight also do not exist - so it does not matter that the initialization branch returned nothing.